### PR TITLE
* Initial commit of snap gene finder recipe

### PR DESCRIPTION
**Pull request by pvanheus**
This adds a recipe for the SNAP ab-initio gene finder (used in MAKER2 amongst other places). 

It is a straightforward build except for one thing: the tools in the snap suite require the ZOE environment variable to point to the location of the HMMs used. To set this environment variable the build script writes to etc/activate.d in the environment where it is installed, an approach thus far only used by the mugsy tool as far as I can see.

### CODE REVIEW

File: recipes/snap/build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+make
+mkdir -p $PREFIX/bin
+mkdir -p ${PREFIX}/share/snap
+mkdir -p ${PREFIX}/etc/conda/activate.d ${PREFIX}/etc/conda/deactivate.d
+cp -p snap fathom forge hmm-info exonpairs cds-trainer.pl hmm-assembler.pl noncoding-trainer.pl patch-hmm.pl zff2gff3.pl $PREFIX/bin
+cp -pr HMM ${PREFIX}/share/snap
+cp -pr DNA ${PREFIX}/share/snap
+
+# set environment variables using conda's environment activate/deactivate
```
Comment (johanneskoester) [inline recipes/snap/build.sh]: Does this also work when installing snap into the root environment?

Comment (pvanheus) [inline recipes/snap/build.sh]: The ZOE environment variable is not set, so no. Any suggestions on how to resolve that? The only other option I can think of for now is replace all binaries with shell script wrappers that set the ZOE environment variable if not set and then call the "real" tool.

Comment (johanneskoester) [inline recipes/snap/build.sh]: Yes, having wrappers that just forward the args via `$@` is the way to go I think.

**Comment by pvanheus** [comment]
As per suggestion from @bgruening and @johanneskoester I've created a wrapper script that sets the ZOE environment variable and then calls the actual SNAP command. Tested and it works.

So this version replaces the old one with its activate / deactivate magic and works for both installing in root and installing in an environment.

File: recipes/snap/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+source:
+  fn: snap-2013-11-29.tar.gz
+  url: http://korflab.ucdavis.edu/Software/snap-2013-11-29.tar.gz
+  #url: https://github.com/amplab/snap/archive/v1.0beta.18.tar.gz
+
+build:
+  number: 0
+
+requirements:
+  build:
```
Comment (bgruening) [inline recipes/snap/meta.yaml]: Peter can you please depend on gcc/llvm is required.

Comment (pvanheus) [inline recipes/snap/meta.yaml]: Ok I've done so for the build.

Comment (bgruening) [inline recipes/snap/meta.yaml]: Sorry to not be clear. We need to specify both gcc and llvm depending on the OS, like here: https://github.com/bioconda/bioconda-recipes/blob/master/recipes/openbabel/2.3.2/meta.yaml#L20

Comment (pvanheus) [inline recipes/snap/meta.yaml]: Ok added the llvm dependency.

**Comment by bgruening** [comment]
Can you merge master into this one. Than I can merge. You can also merge if you like. It looks good to me! Thanks a lot!

**Comment by pvanheus** [comment]
Is the "Update" I did from master the "merge master" that you mentioned? I'm still learning the Github ropes...

**Comment by bgruening** [comment]
One last idea; Can you import `- pkg-config` as build dep? I will have time tm to try by myself, but zlib can be really ugly. I wish I would know why this is so special.

**Comment by bgruening** [comment]
Yes, thanks @pvanheus. I just clicked the wrong button know .. arg :)

**Comment by bgruening** [comment]
@johanneskoester can you please merge this. I missed the short period where it was possible.
@pvanheus You can also merge it by your own, if you got commit access.

**Comment by johanneskoester** [comment]
@bgruening I just discovered that there is an option to deactivate these nasty requirement that the branch to merge is up to date with master. Now, any conflict free merge can be done by any dev! Let's try this for the next days and see if it works properly.

**Comment by bgruening** [comment]
@johanneskoester awesome! Let's try it :)

**Pull request merged**